server/src/main/java/org/opensearch/action/admin/indices/create/CreateIndexRequest.java

```diff
@@ -229,7 +229,9 @@ public CreateIndexRequest settings(Map<String, ?> source) {
      * @param type   The mapping type
      * @param source The mapping source
      * @param xContentType The content type of the source
+     * @deprecated types are being removed
```

Comment:
Why we are keeping the types in CreateIndexRequest ?

Reply:
Because removing these is a large change that will need to come incrementally to maintain bwc due to the serialization dependency.

Reply:
Thanks for the clarification